cli/src/cmd/forge/script/mod.rs

```diff
@@ -699,6 +713,30 @@ impl ScriptConfig {
     fn target_contract(&self) -> &ArtifactId {
         self.target_contract.as_ref().expect("should exist after building")
     }
+
+    /// Checks if the RPCs used point to chains that support EIP-3855.
+    /// If not, warns the user.
+    async fn check_shanghai_support(&self) -> eyre::Result<()> {
+        for rpc in &self.total_rpcs {
+            let provider = ethers::providers::Provider::<Http>::try_from(rpc)?;
+            let chain_id = provider.get_chainid().await?;
+            if !SHANGHAI_ENABLED_CHAINS.contains(&chain_id.try_into()?) {
```

Comment:
These should be `let Ok(...) = ... else { continue };` to continue instead of returning and ignoring all other rpcs